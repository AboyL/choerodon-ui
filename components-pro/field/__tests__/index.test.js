import React from 'react';
import { mount } from 'enzyme';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import ObserverFormField from '..';

@observer
class TestFiledForm extends ObserverFormField {
  renderWrapper() {
    return (
      <div>
        <input />
      </div>
    );
  }
}

describe('Select', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('observer component reder right', () => {
    const wrapper = mount(<TestFiledForm />);
    jest.runAllTimers();
    expect(wrapper.exists('input')).toMatchSnapshot();
  });
  it('onclick should tobe call', () => {
    const mockFn = jest.fn();
    @observer
    class TestFiledForm1 extends ObserverFormField {
      static propTypes = {
        onClick: PropTypes.func,
      };

      renderWrapper() {
        return (
          <div onClick={this.props.onClick}>
            <input />
          </div>
        );
      }
    }
    const wrapper = mount(<TestFiledForm1 onClick={mockFn} />);
    wrapper.simulate('click');
    jest.runAllTimers();
    expect(mockFn).toHaveBeenCalled();
  });
});
