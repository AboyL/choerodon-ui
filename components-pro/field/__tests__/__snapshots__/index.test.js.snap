// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select observer component reder right 1`] = `true`;
